src/main/java/io/vertx/proton/ProtonClientOptions.java

```diff
@@ -314,6 +324,48 @@ public ProtonClientOptions setLocalAddress(String localAddress) {
   }
 
   /**
+   * Override the host to use in the connection open frame. If this is not set, the hostname specified in
+   * {@link ProtonClient#connect} will be used.
+   *
+   * @param virtualHost hostname to set
+   * @return  current ProtonClientOptions instance
+   */
+  public ProtonClientOptions setVirtualHost(String virtualHost) {
+    this.virtualHost = virtualHost;
+    return this;
+  }
+
+  /**
+   * Return the host override for the connection open frame.
+   *
+   * @return  the hostname
+   */
+  public String getVirtualHost() {
+    return this.virtualHost;
+  }
+
+  /**
+   * Override the host to use for the SNI Server Name. If this is not set, the hostname specified in
```

Comment:
Should mention the virtualhost option interaction, making it clearer this option will only be needed in rare cases.